src/frontend/src/instance.rs

```diff
@@ -455,6 +455,17 @@ impl PrometheusHandler for Instance {
     }
 }
 
+/// Validate `stmt.database` permission if it's presented.
```

Comment:
Is it possible to be a method? I don't find it have to be a macro.

Reply:
Because the `stmt` has different types. Of course, it can be changed as a function such as `validate_db_permission(&stmt.database, query_ctc)?`.

But looks not so simple as this macro.

Reply:
Make sense.